src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -47,66 +47,102 @@ private static void FreeDeconstructionVariables(ArrayBuilder<DeconstructionVaria
+                                                        DiagnosticBag diagnostics,
+                                                        bool isDeclaration,
+                                                        BoundDeconstructValuePlaceholder rhsPlaceholder = null)
         {
-            TypeSymbol voidType = GetSpecialType(SpecialType.System_Void, diagnostics, node);
-
             // receiver for first Deconstruct step
             var boundRHS = rhsPlaceholder ?? BindValue(right, diagnostics, BindValueKind.RValue);
 
+            boundRHS = FixTupleLiteral(checkedVariables, boundRHS, node, right, diagnostics);
+
             if ((object)boundRHS.Type == null)
             {
-                if (boundRHS.Kind == BoundKind.TupleLiteral)
-                {
-                    // Let's fix the literal up by figuring out its type
-                    // For declarations, that means merging type information from the LHS and RHS
-                    // For assignments, only the LHS side matters since it is necessarily typed
-                    TypeSymbol lhsAsTuple = MakeMergedTupleType(checkedVariables, (BoundTupleLiteral)boundRHS, node, Compilation, diagnostics);
-                    if ((object)lhsAsTuple != null)
-                    {
-                        boundRHS = GenerateConversionForAssignment(lhsAsTuple, boundRHS, diagnostics);
-                    }
-                }
-                else
-                {
-                    Error(diagnostics, ErrorCode.ERR_DeconstructRequiresExpression, right);
-                }
-
-                if ((object)boundRHS.Type == null)
-                {
-                    // we could still not infer a type for the RHS
-                    FailRemainingInferences(checkedVariables, diagnostics);
-
-                    return new BoundDeconstructionAssignmentOperator(
-                                node, isDeclaration, FlattenDeconstructVariables(checkedVariables), boundRHS,
-                                ImmutableArray<BoundDeconstructionDeconstructStep>.Empty, ImmutableArray<BoundDeconstructionAssignmentStep>.Empty,
-                                voidType, hasErrors: true);
-                }
+                // we could still not infer a type for the RHS
+                FailRemainingInferences(checkedVariables, diagnostics);
+
+                return new BoundDeconstructionAssignmentOperator(
+                            node, isDeclaration, FlattenDeconstructVariables(checkedVariables), boundRHS,
+                            ImmutableArray<BoundDeconstructionDeconstructStep>.Empty,
+                            ImmutableArray<BoundDeconstructionAssignmentStep>.Empty,
+                            ImmutableArray<BoundDeconstructionAssignmentStep>.Empty,
+                            ImmutableArray<BoundDeconstructionConstructionStep>.Empty,
+                            GetSpecialType(SpecialType.System_Void, diagnostics, node),
+                            hasErrors: true);
```

Comment:
The tests that cover this path (failed inference) do not have any diagnostics related to the void type that is returned here.